src/sync_template.sh

```diff
@@ -243,11 +251,30 @@ function pull_source_changes() {
 
   eval "git pull ${source_repo} --tags ${git_remote_pull_params}" || pull_has_issues=true
 
+  info "finished pulling from the source."
+  info "logging out from source ${SOURCE_REPO_HOSTNAME}."
+
+  if [[ -n "${SRC_SSH_PRIVATEKEY_ABS_PATH}" ]] &>/dev/null; then
+    info "we are using ssh for the source repo. No need to logout."
+  else
+    gh auth status --hostname "${SOURCE_REPO_HOSTNAME}" 
+    gh auth status --active    
+
+    # Clear the GH_TOKEN environment variable
+    unset GH_TOKEN
+
+    info "logging in into the target with hostname ${GITHUB_SERVER_URL}"
+    gh auth login --git-protocol "https" --hostname "github.com" --with-token <<<"${TARGET_GH_TOKEN}"
+    gh auth switch --hostname "${GITHUB_SERVER_URL}"   
```

Comment:
If I am not mistaken the `GITHUB_SERVER_URL`  contains the hostname with the path. Is this working?

Reply:
I am currently trying things out. Looks not like it is working as I would like it to be

Reply:
::info::logging in into the target with hostname https://github.com
  not logged in to https://github.com 

  I think it the variable is not filled correctly. and that is why the switch can not work either as we are not logged into https://github.com